encryption/encryption.cc

```diff
@@ -457,7 +461,12 @@ Status KeyManagedEncryptedEnv::RenameFile(const std::string& src_fname,
                                           const std::string& dst_fname) {
   if (ShouldSkipEncryption(dst_fname)) {
     assert(ShouldSkipEncryption(src_fname));
-    return target()->RenameFile(src_fname, dst_fname);
+    Status s = target()->RenameFile(src_fname, dst_fname);
+    // Replacing with plaintext requires deleting the info in the key manager.
+    Status delete_status __attribute__((__unused__)) =
+        key_manager_->DeleteFile(dst_fname);
```

Comment:
This is only needed when `ShouldSkipEncryption(dst_fname)` returns false once. How can this happen?

Reply:
when upgrade from old version like https://github.com/tikv/tikv/issues/9701

Reply:
@hunterlxt update the inline comment to mention it is for upgrading from TiKV <= v5.0.0-rc?